packages/react/src/hooks/useTimeout.spec.tsx

```diff
@@ -16,6 +16,25 @@ describe('useTimeout', () => {
     expect(spy).toHaveBeenCalledTimes(1)
   })
 
+  it('should call the modified function when the given function is changed', async () => {
+    const timeoutMockFn1 = vi.fn()
+    const timeoutMockFn2 = vi.fn()
+
+    const { rerender } = renderHook(({ fn }) => useTimeout(fn, ms('0.1s')), {
+      initialProps: { fn: timeoutMockFn1 },
+    })
+
+    rerender({ fn: timeoutMockFn2 })
+
+    expect(timeoutMockFn1).toHaveBeenCalledTimes(0)
+    expect(timeoutMockFn2).toHaveBeenCalledTimes(0)
+
+    await sleep(ms('0.1s'))
+
+    expect(timeoutMockFn1).toHaveBeenCalledTimes(0)
+    expect(timeoutMockFn2).toHaveBeenCalledTimes(1)
+  })
```

Comment:
If it's existing code, it won't call the changed function as intended, and the test will fail. 😭